syntheseus/search/analysis/route_extraction.py

```diff
@@ -189,3 +189,40 @@ def min_cost_routes(
             break
         else:
             yield route
+
+
+def _route_time_cost(nodes, graph) -> float:
+    """
+    Cost function for routes that is the maximum timestamp of any node in the route
+    (or inf if it is not solved).
+    """
+    if _route_has_solution(nodes, graph):
+        return max(n.creation_time.timestamp() for n in nodes)
+    else:
+        return math.inf
+
+
+def _route_time_partial_cost(nodes, _) -> float:
+    """Partial cost version of above. It is a lower bound to the true cost."""
+    if all(n.has_solution for n in nodes):
+        return max(n.creation_time.timestamp() for n in nodes)
+    else:
+        return math.inf
+
+
+def iter_routes_in_order_found(
+    graph: RetrosynthesisSearchGraph, max_routes: int
+) -> Iterator[Collection[BaseGraphNode]]:
+    """
+    Yield all routes from "graph" in the order they were found.
+    Graph can be AND/OR or MolSet graph. Order is determined by "creation_time".
```

Comment:
nit: If you are going to put something around variable names or other symbols, then I think the standard thing is to use backticks like [here](https://github.com/microsoft/syntheseus/blob/e255b0559286aee34584742dc980e68b60f6cae2/syntheseus/search/graph/message_passing/run.py#L34), e.g.
```suggestion
    Yield all routes from `graph` in the order they were found.
    Graph can be AND/OR or MolSet graph. Order is determined by `creation_time`.
```
But while some parts of `syntheseus` use backticks, other parts don't use anything, so it's not too consistent at the moment.

Reply:
Yeah, I haven't been very consistent, my bad. Which would you prefer? Backticks or quotes?

Reply:
I personally use backticks quite consistently (and so this is what you'll find in e.g. single-step code), but usually don't enforce it on others too aggressively 🙂 After a quick look, there are cases in `syntheseus` that use backticks, quotes, or nothing at all, but maybe if we decide on what we prefer then this will slowly converge one way or another in future PRs.

I think this stuff matters more when one uses an automated tool to convert docstrings into documentation, in which case backticks are likely to be rendered specially. Otherwise it's maybe slightly nicer to read as mentally one knows when to expect a variable/symbol.

Reply:
Sounds good. I'm happy with this approach.